cpp/src/traversal/sssp.cu

```diff
@@ -314,91 +310,42 @@ void sssp(Graph* graph,
     // BFS also does only integer distances right now whereas we need float or
     // double
 
-    void* d_edge_data;
-    graph->adjList->edge_data = new gdf_column;
     cudaStream_t stream{nullptr};
-
-    // If distances array is given and is double, generate the weights in
-    // double
-    if (distances && typeid(WT) == typeid(double)) {
-      std::vector<double> h_edge_data(graph->adjList->indices->size, 1.0);
-      size_t edge_data_size = sizeof(double) * h_edge_data.size();
-      ALLOC_TRY((void**)&d_edge_data, edge_data_size, stream);
-      CUDA_TRY(cudaMemcpy(d_edge_data,
-                          &h_edge_data[0],
-                          edge_data_size,
-                          cudaMemcpyHostToDevice));
-      gdf_column_view(graph->adjList->edge_data,
-                      d_edge_data,
-                      nullptr,
-                      graph->adjList->indices->size,
-                      GDF_FLOAT64);
-
-    } else {
-      // Else generate float
-      std::vector<float> h_edge_data(graph->adjList->indices->size, 1.0);
-      size_t edge_data_size = sizeof(float) * h_edge_data.size();
-      ALLOC_TRY((void**)&d_edge_data, edge_data_size, stream);
-      CUDA_TRY(cudaMemcpy(d_edge_data,
-                          &h_edge_data[0],
-                          edge_data_size,
-                          cudaMemcpyHostToDevice));
-      gdf_column_view(graph->adjList->edge_data,
-                      d_edge_data,
-                      nullptr,
-                      graph->adjList->indices->size,
-                      GDF_FLOAT32);
-    }
+    std::vector<WT> h_edge_data(graph.number_of_edges, static_cast<WT>(1));
+    size_t edge_data_size = sizeof(WT) * h_edge_data.size();
+    ALLOC_TRY((WT**)&edge_weights_ptr, edge_data_size, stream);
```

Comment:
Try to use rmm::device_vector or device_buffer here

Reply:
Updated in https://github.com/rapidsai/cugraph/commit/1528aebe9ea39d6ac0060b1068a76f8c2a36056c, uses thrust::device_vector instead